packages/core/src/indexing/service.ts

```diff
@@ -527,264 +712,277 @@ export class IndexingService extends Emittery<IndexingEvents> {
+          (s) =>
+            s.contractName === contractName &&
+            s.abiEvents.bySafeName[eventName] !== undefined,
+        );
+
+        const abiEvent = this.sources[i].abiEvents.bySafeName[eventName]!.item;
+        const eventSelector =
+          this.sources[i].abiEvents.bySafeName[eventName]!.selector;
+
+        this.indexingFunctionMap[indexingFunctionKey] = {
+          eventName,
+          contractName,
+          parents: dedupe(parents),
+          selfReliance,
+          sources: keySources,
+          abiEvent,
+          eventSelector,
+
+          checkpoint: zeroCheckpoint,
+          maxTaskCheckpoint: zeroCheckpoint,
+          indexingFunctionTasks: [],
+          dbMutex: new Mutex(),
+          serialQueue: false,
+        };
+      }
+    }
+  };
 
-  networks.forEach((network, i) => {
-    const defaultChain =
-      Object.values(chains).find((c) => c.id === network.chainId) ??
-      chains.mainnet;
-
-    const client = createClient({
-      transport: ponderTransport({
-        requestQueue: requestQueues[i],
-        syncStore,
-      }),
-      chain: { ...defaultChain, name: network.name, id: network.chainId },
+  private setEventMetrics = async () => {
+    const { counts } = await this.syncGatewayService.getEventCounts({
+      fromCheckpoint: zeroCheckpoint,
+      toCheckpoint: this.syncGatewayService.checkpoint,
+      logFilters: this.sources.filter(sourceIsLogFilter).map((logFilter) => ({
+        id: logFilter.id,
+        chainId: logFilter.chainId,
+        criteria: logFilter.criteria,
+        fromBlock: logFilter.startBlock,
+        toBlock: logFilter.endBlock,
+      })),
+      factories: this.sources.filter(sourceIsFactory).map((factory) => ({
+        id: factory.id,
+        chainId: factory.chainId,
+        criteria: factory.criteria,
+        fromBlock: factory.startBlock,
+        toBlock: factory.endBlock,
+      })),
     });
 
-    contexts[network.chainId] = {
-      network: { name: network.name, chainId: network.chainId },
-      // Changing the arguments of readContract is not usually allowed,
-      // because we have such a limited api we should be good
-      client: client.extend(actions as any) as ReadOnlyClient,
-      contracts: {},
-    };
-  });
-
-  sources.forEach((source) => {
-    // Only include the address if it's singular and  static.
-    const address =
-      typeof source.criteria.address === "string"
-        ? source.criteria.address
-        : undefined;
-
-    contexts[source.chainId] = {
-      ...contexts[source.chainId],
-      contracts: {
-        ...contexts[source.chainId].contracts,
-        [source.contractName]: {
-          abi: source.abi,
-          address: address ? checksumAddress(address) : address,
-          startBlock: source.startBlock,
-          endBlock: source.endBlock,
-          maxBlockRange: source.maxBlockRange,
-        },
-      },
-    };
-  });
+    for (const count of counts) {
+      const event =
```

Comment:
Would consider doing this here:
`const source = this.sourceById[count.sourceId]!`